File: Daily Test/Java/Match Left And Right - Words 12.1.2026.java
import java.util.*;

public class Hello {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String[] L = sc.nextLine().split(" ");
        String[] R = sc.nextLine().split(" ");

        boolean[] used = new boolean[R.length];
        StringBuilder out = new StringBuilder();

        for (String l : L) {
            char ch = l.charAt(l.length() - 1);
            for (int i = 0; i < R.length; i++) {
                if (!used[i] && R[i].charAt(0) == ch) {
                    out.append(l).append(R[i].substring(1)).append(" ");
                    used[i] = true;
                    break;
                }
            }
        }
        System.out.print(out.toString().trim());
    }
}

File: Daily Test/Python/Match Left And Right - Words 12.1.2026.py
s1=input().split()
s2=input().split()
used=[0]*len(s2)
ans=[]
for a in s1:
    for i,b in enumerate(s2):
        if not used[i] and a[-1]==b[0]:
            ans.append(a+b[1:])
            used[i]=1
            break
print(*ans)